Sampled/UI/Models/LibraryModel.swift

```diff
@@ -849,7 +837,26 @@ final class LibraryModel {
         var iterator = tracks.makeIterator()
 
         if let track = iterator.next() {
-          var position = try generateKeyBetween(a: track.library.currentQueue?.items.last?.item.position, b: nil)
+          // Get the latest position from the current queue if it exists
+          let latestPosition: String? = if let queueID = track.library.library.currentQueue {
+            // Query for the item with the highest position in this queue using a SQL join
```

Comment:
Is there any way we can roll this into the query above? If not, can you at least use the format like above, given that yours uses raw SQL.

Reply:
Use GRDB's DSL format to code the SQL in Swift securely.